### PR TITLE
Add tests for regional health check deletions

### PR DESCRIPTION
**Pull request by skmatti**
/assign @freehan @bowei

### CODE REVIEW

File: pkg/healthchecks/healthchecks_test.go [hunk 1]
```diff
@@ -269,6 +271,57 @@ func TestHTTP2HealthCheckDelete(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("Unexpected error creating composite key: %v", err)
+	}
+
+	// Verify that Health-check exists.
+	existingHc, err := composite.GetHealthCheck(fakeGCE, key, features.L7ILBVersions().HealthCheck)
+	if err != nil {
+		t.Fatalf("GetHealthCheck(%q) = %v, want nil", hc.Name, err)
+	}
+	if existingHc == nil {
```
Comment (bowei) [inline pkg/healthchecks/healthchecks_test.go]: combine with above:

```
if err != nil || existingHC == nil {
  t.Fatalf("GetHealthCheck(%q) = %v, %v, want nil", hc.Name, err, existingHC)
}
```

**Comment by skmatti** [comment]
> would this test have caught the bug that you fixed?

Yes, the delete would return non-nil error trying to delete global health-check.

**Review by freehan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**